src/openrct2-ui/windows/Window.h

```diff
@@ -168,3 +168,17 @@ void window_tooltip_reset(ScreenCoordsXY screenCoords);
 void window_tooltip_show(rct_string_id id, ScreenCoordsXY screenCoords);
 void window_tooltip_open(rct_window* widgetWindow, rct_widgetindex widgetIndex, ScreenCoordsXY screenCoords);
 void window_tooltip_close();
+
+#ifndef LEFT_CLOSEBOX
+    // close button on right of window
+    #define WINDOW_SKELETON \
+        { WWT_FRAME,    0,  0,  WW - 1, 0,  WH, 0xFFFFFFFF,                                 STR_NONE }, \
```

Comment:
This is indeed roughly what I had in mind. I'd like to see two things changed:
- `WINDOW_SKELETON` should be a "function" that takes the window name, width and height: `#define WINDOW_SKELETON(title, width, height) \` and change the body accordingly.
- On the old `WIDGETS_MAIN`, we deliberately left out the first and last brace, so that we could use `{ WIDGETS_MAIN }` in our window definitions. Please replicate that.

When you apply these changes, the window definitions themselves will look like this:
```
static rct_widget window_staff_overview_widgets[] = {
    { WINDOW_SKELETON(STR_STRINGID, WW, WH) },
    { WWT_RESIZE,   1, 0,       WW - 1,     43,         WH - 1, 0xFFFFFFFF,             STR_NONE },             // Resize
```